gordo_components/builder/mlflow_utils.py

```diff
@@ -474,20 +477,9 @@ def log_metadata(mlflow_client, run_id, metadata: dict):
 
     # Send configs as JSON artifacts
     try:
-        with tempfile.TemporaryDirectory() as tmp_dir:
-            for name, obj in [
-                ("model-config", metadata["model"]),
-                ("dataset-config", metadata["dataset"]),
-                (
-                    "user-config",
-                    metadata.get("user-defined") or metadata.get("metadata"),
-                ),
-            ]:
-                fp = os.path.join(tmp_dir, f"{name}.json")
-                json.dump(obj, open(fp, "w"), cls=DatetimeEncoder)
-            mlflow_client.log_artifacts(run_id, local_dir=tmp_dir)
-    # Log error without exiting the build
-    # FIX: when a solution is found when logging artifacts wit azureml lib,
-    # this can raise an error
+        with tempfile.TemporaryDirectory(dir="./") as tmp_dir:
+            fp = os.path.join(tmp_dir, f"metadata.json")
+            json.dump(metadata, open(fp, "w"), cls=DatetimeEncoder)
```

Comment:
Is it safe not to use a context manager when writing to this file, do you know if the internals of `json.dumps` will close the file if it cannot complete serialization process?

Reply:
What do you mean by serialization process?

Reply:
if `json.dumps` (the process of converting `metadata` to json) fails, there is nothing in there that will close the handle to this file.

Reply:
As discussed, rolling this one into #795.